### PR TITLE
Upgrade operator-sdk from 0.8.2 to 0.17.1

### PR DESCRIPTION
**Pull request by 2uasimojo**
This is the latest level [currently supported by boilerplate](https://github.com/openshift/boilerplate/blob/d5accbb1fb8654c00f3cfc1f4395e2ae7f493ad4/boilerplate/openshift/golang-osd-operator/operator-sdk-generate.sh#L23).

Jira: [OSD-5370](https://issues.redhat.com/browse/OSD-5370)

### CODE REVIEW

File: standard.mk [hunk 1]
```diff
@@ -27,11 +27,12 @@ OPERATOR_DOCKERFILE ?=build/ci-operator/Dockerfile
...
 MAINPACKAGE=./cmd/manager
-GOENV=GOOS=linux GOARCH=amd64 CGO_ENABLED=0
-GOFLAGS=-gcflags="all=-trimpath=${GOPATH}" -asmflags="all=-trimpath=${GOPATH}" -mod=vendor
+unexport GOFLAGS
+GOENV=GOOS=linux GOARCH=amd64 CGO_ENABLED=0 GOFLAGS=
+GOBUILDFLAGS=-gcflags="all=-trimpath=${GOPATH}" -asmflags="all=-trimpath=${GOPATH}"
 export GO111MODULE=on
 
-TESTTARGETS := $(shell go list -e ./... | egrep -v "/(vendor)/")
+TESTTARGETS := $(shell go list -e ./...)
```
Comment (cblecker) [inline standard.mk:35]: Remove this line. Replace the var on L74 with the literal `./...`

Comment (2uasimojo) [reply standard.mk]: Done.

File: cmd/manager/main.go [hunk 2]
```diff
@@ -108,7 +114,12 @@ func run() error {
 	}
 
 	log.V(2).Info("Expose metrics")
-	_, err = metrics.ExposeMetricsPort(ctx, metricsPort)
+	// TODO: Use operator-custom-metrics
+	servicePorts := []v1.ServicePort{
+		{Port: metricsPort, Name: metrics.OperatorPortName, Protocol: v1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
+	}
+
+	_, err = metrics.CreateMetricsService(ctx, cfg, servicePorts)
```
Comment (2uasimojo) [inline cmd/manager/main.go:122]: @fahlmant Should we even bother with this, or just blow this whole section away for now?

File: cmd/manager/main.go [hunk 3]
```diff
@@ -87,7 +93,7 @@ func run() error {
 	// Create a new Cmd to provide shared dependencies and start components
 	mgr, err := manager.New(cfg, manager.Options{
 		Namespace:          "", //watch all namespaces
-		MapperProvider:     restmapper.NewDynamicRESTMapper,
+		MapperProvider:     func(cfg *rest.Config) (meta.RESTMapper, error) { return apiutil.NewDynamicRESTMapper(cfg) },
```
Comment (2uasimojo) [inline cmd/manager/main.go:96]: Note to reviewers: The [docs](https://sdk.operatorframework.io/docs/upgrading-sdk-version/version-upgrade-guide/#v014x) said s/restmapper/apiutil/ was a drop-in replacement. They lied.

File: deploy/cluster_role.yaml [hunk 4]
```diff
@@ -15,7 +15,13 @@ rules:
   - configmaps
   - secrets
   verbs:
-  - '*'
+  - create
```
Comment (2uasimojo) [inline deploy/cluster_role.yaml:18]: Note to reviewers: Expanding these verbs per the [docs](https://sdk.operatorframework.io/docs/upgrading-sdk-version/version-upgrade-guide/#v013x) -- see also https://github.com/operator-framework/operator-sdk/issues/671

**Comment by 2uasimojo** [comment]
@georgettica helped me test this today. Thanks Ron!

**Comment by georgettica** [comment]
I tested with @2uasimojo on my crc and we got the operator up and running. when possible I recommend merging 👍

File: cmd/manager/main.go (hunk 2 above)
Comment (fahlmant) [reply cmd/manager/main.go:122]: This is the operator-sdk metrics package, which I don't have experience with

File: deploy/crds/gcp.managed.openshift.io_projectclaims_crd.yaml [hunk 5]
```diff
@@ -128,16 +138,18 @@ spec:
...
                 type: object
               type: array
             state:
+              description: ClaimStatus is a valid value from ProjectClaim.Status
               type: string
           required:
           - conditions
           - state
           type: object
+      type: object
```
Comment (NautiluX) [inline deploy/crds/gcp.managed.openshift.io_projectclaims_crd.yaml:152]: in the past we have seen problems deploying to v3.11 with this additional `type: object`, @georgettica the solution back then was to just remove it here right?

File: deploy/crds/gcp.managed.openshift.io_projectreferences_crd.yaml [hunk 6]
```diff
@@ -113,16 +122,18 @@ spec:
...
                 type: object
               type: array
             state:
+              description: ProjectReferenceState is a valid value from ProjectReference.Status
               type: string
           required:
           - conditions
           - state
           type: object
+      type: object
```
Comment (NautiluX) [inline deploy/crds/gcp.managed.openshift.io_projectreferences_crd.yaml:136]: see above

File: deploy/crds/gcp.managed.openshift.io_projectclaims_crd.yaml (hunk 5 above)
Comment (georgettica) [reply deploy/crds/gcp.managed.openshift.io_projectclaims_crd.yaml:152]: yup, if its the last one in the file then you.

all we need to do is deploy the CRD on a v3 cluster and see if it holds, anything else should work without a hitch

File: deploy/crds/gcp.managed.openshift.io_projectreferences_crd.yaml (hunk 6 above)
Comment (georgettica) [reply deploy/crds/gcp.managed.openshift.io_projectreferences_crd.yaml:136]: same response

File: deploy/crds/gcp.managed.openshift.io_projectclaims_crd.yaml (hunk 5 above)
Comment (2uasimojo) [reply deploy/crds/gcp.managed.openshift.io_projectclaims_crd.yaml]: Resolved via 488f569.

File: deploy/crds/gcp.managed.openshift.io_projectreferences_crd.yaml (hunk 6 above)
Comment (2uasimojo) [reply deploy/crds/gcp.managed.openshift.io_projectreferences_crd.yaml]: Resolved via 488f569.

**Review by NautiluX — APPROVED** [review]
/lgtm
/approve

feel free to `unhold` as soon as the v3 tests verify it's working

**Comment by georgettica** [comment]
/hold
need to run a `make` to verifty it's still ok

**Comment by georgettica** [comment]
@2uasimojo I tried to make it work, can you ack so we can continue merging?

**Comment by 2uasimojo** [comment]
We’re ready to go here. 

/hold cancel

**Pull request merged**